src/template.c

```diff
@@ -1,11 +1,20 @@
-#include <Rcpp.h>
-using namespace Rcpp;
+#include <Rinternals.h>
+
+SEXP str_sxp_set(SEXP x, int i, SEXP val) {
+  R_xlen_t len = Rf_xlength(x);
+  if (i >= len) {
+    len *= 2;
+    x = Rf_lengthgets(x, len);
```

Comment:
Does this need to be protected? (Presumedly we have to use `REPROTECT()`)

Reply:
The REPROTECT happens outside this function. I think this should be safe because `SET_STRING_ELT` won't allocate, and rchk doesn't have issue with this code. But it probably would be safer to pass the protection index to this function and just do the `REPROTECT` inside. I guess also currently we are REPROTECTing more often than we need to, I will change it.

Reply:
So actually what you suggest is how I had written this code originally (https://github.com/tidyverse/glue/commit/b0cfea86bc80856ef56cdae05532573ecbc35e47#diff-92efcf7db90903ecdc03a20503b14360) and unfortunately rchk has a false positive for this, it cannot track REPROTECTs across function boundaries. However converting the function to a macro solves the issue and also makes the code look more like a simple `SET_STRING_ELT()`, which is what I have done now.